### PR TITLE
XW-3939 | Update Ember

### PR DESCRIPTION
**Pull request by hakubo**
## Links

* XW-3939

## Description

Ember updated to 2.15
Ember cli update to 2.15
Fastboot to 1.0.5 (our patch for shoebox appendto reapplied)

## Reviewers

Use `@` mentions here. Alternatively, assign the pull request to a person.

### CODE REVIEW

File: app/components/article-comment.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Ember from 'ember';
 import Thumbnailer from '../modules/thumbnailer';
 
-const {Component, computed, inject} = Ember;
+const {$, Component, computed, inject} = Ember;
```
Comment (latata) [inline app/components/article-comment.js]: `$` was not needed before?

File: app/routes/wiki-page.js [hunk 2]
```diff
@@ -159,8 +159,8 @@ export default Route.extend(
 					}
 
 					if (fastboot.get('isFastBoot')) {
-						fastboot.get('response.headers').set('location', redirectTo);
-						fastboot.set('response.statusCode', 301);
+						//fastboot.get('response.headers').set('location', redirectTo);
```
Comment (latata) [inline app/routes/wiki-page.js]: why is that commented out?

File: app/components/article-media-thumbnail.js [hunk 3]
```diff
@@ -14,8 +16,6 @@ export default Ember.Component.extend(
 		classNameBindings: ['itemType', 'isLoading', 'isSmall'],
 		tagName: 'figure',
 
-		layoutName: 'components/article-media-thumbnail',
```
Comment (rogatty) [inline app/components/article-media-thumbnail.js:17]: Are you 100% sure it's not needed anymore? IIRC there were issues in galleries without it.

File: app/initializers/link-to.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+import {track, trackActions} from '../utils/track';
+
+const {LinkComponent} = Ember;
+
+/**
+ * @returns {void}
+ */
+export function initialize() {
+	if (typeof FastBoot === 'undefined') {
```
Comment (rogatty) [inline app/initializers/link-to.js]: We could reduce indentation by using early return instead:
```
if (typeof FastBoot !== 'undefined') {
    return;
}
```

File: app/inline-scripts/load-svg.js [hunk 5]
```diff
@@ -4,26 +4,21 @@
...
 
-		return template.firstChild;
+		return template.querySelector('svg');
 	}
 
 	function loadDOMResource(src) {
-		var ajax = new XMLHttpRequest();
+		fetch(src)
+			.then(function(response) {
+				return response.text();
```
Comment (rogatty) [inline app/inline-scripts/load-svg.js]: What happens when it throws? You can check it by blocking svg network request in DevTools.

File: app/router.js [hunk 6]
```diff
@@ -3,7 +3,8 @@ import RouterScroll from 'ember-router-scroll';
 import config from './config/environment';
 
 const Router = Ember.Router.extend(RouterScroll, {
-	location: config.locationType
+	location: config.locationType,
+	rootURL: config.rootURL
```
Comment (rogatty) [inline app/router.js:7]: I remember we had some issues when using `rootURL`, maybe they're no longer valid. Did you test it on sandbox? I think it was something about fingerprinting, so please check the file names if they're as expected.

File: app/routes/application.js [hunk 7]
```diff
@@ -143,26 +142,6 @@ export default Route.extend(
 			}
 		},
 
-		redirect(model, transition) {
```
Comment (rogatty) [inline app/routes/application.js:146]: I guess this is still in progress. Please don't forget about reimplementing.

File: app/routes/wiki-page.js (hunk 2 above)
Comment (rogatty) [reply app/routes/wiki-page.js]: I guess this is still in progress. Please don't forget about reimplementing.

File: ember-cli-build.js [hunk 8]
```diff
@@ -14,25 +10,25 @@ var EmberApp = require('ember-cli/lib/broccoli/ember-app'),
...
+// 		if (addon.treeFor) {
+// 			let tree = addon.treeFor(type);
+//
+// 			if (tree) {
+// 				// uncomment to see the files available to be filtered out
+// 				// tree = stew.log(tree, {output: 'tree'});
+// 				tree = stew.rm(tree,
+// 					'modules/ember-types/asserts/**/*.js',
+// 					'modules/ember-types/constants/*.js',
+// 					'modules/ember-types/property/*.js'
```
Comment (rogatty) [inline ember-cli-build.js]: There is a lot of unnecessary code in these files. If this way doesn't work anymore, is there something else we can do?

File: app/utils/url.js [hunk 9]
```diff
@@ -125,7 +125,7 @@ export function buildUrl(urlParams = {}) {
 }
 
 export function extractEncodedTitle(url) {
-	return url ? url.replace(/^(http:\/\/[^\/]+)?(\/wiki)?\//, '') : '';
```
Comment (rogatty) [inline app/utils/url.js:128]: Are you sure about this? How is it related to the update?

File: app/utils/string.js [hunk 10]
```diff
@@ -49,5 +49,5 @@ export function getLastUrlFromText(text) {
  * @return {string}
  */
 export function escapeRegex(text) {
-	return text.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
```
Comment (rogatty) [inline app/utils/string.js:52]: Are you sure about this? How is it related to the update?

File: app/routes/application.js (hunk 7 above)
Comment (hakubo) [reply app/routes/application.js:146]: do you remember what exactly this was handling?

Comment (rogatty) [reply app/routes/application.js:146]: http://the-teen-wolf.wikia.com/wiki/Scott_McCall redirect to http://teen-wolf-pack.wikia.com/wiki/Scott_McCall (primary domain)

File: app/components/article-comment.js (hunk 1 above)
Comment (hakubo) [reply app/components/article-comment.js]: it was using global $

File: app/routes/application.js (hunk 7 above)
Comment (rogatty) [reply app/routes/application.js:146]: Related: https://wikia-inc.atlassian.net/browse/XW-3198

File: app/components/article-media-thumbnail.js (hunk 3 above)
Comment (hakubo) [reply app/components/article-media-thumbnail.js:17]: It's injected in render-component,

File: app/inline-scripts/load-svg.js [hunk 11]
```diff
@@ -4,26 +4,21 @@
 
 		template.insertAdjacentHTML('beforeend', html);
 
-		return template.firstChild;
+		return template.querySelector('svg');
 	}
 
 	function loadDOMResource(src) {
-		var ajax = new XMLHttpRequest();
+		fetch(src)
```
Comment (hakubo) [inline app/inline-scripts/load-svg.js:11]: it runs after vendor
so ember-fetch is already loaded

File: app/utils/url.js (hunk 9 above)
Comment (hakubo) [reply app/utils/url.js:128]: updated eslint rules, no-useless-escape

File: app/utils/string.js (hunk 10 above)
Comment (hakubo) [reply app/utils/string.js:52]: updated eslint rules, no-useless-escape

File: app/routes/application.js (hunk 7 above)
Comment (hakubo) [reply app/routes/application.js:146]: ok, reapplied

File: app/routes/wiki-page.js [hunk 12]
```diff
@@ -159,17 +165,12 @@ export default Route.extend(
...
 						});
 					}
 
-					if (fastboot.get('isFastBoot')) {
-						fastboot.get('response.headers').set('location', redirectTo);
-						fastboot.set('response.statusCode', 301);
-					} else {
-						window.location.replace(redirectTo);
-					}
+					this.replaceWith(redirectTo);
```
Comment (hakubo) [inline app/routes/wiki-page.js]: test if that still works

File: app/styles/component/_wikia-drawer.scss [hunk 13]
```diff
@@ -1,4 +1,4 @@
-&.collapsed .side-nav-drawer {
```
Comment (hakubo) [inline app/styles/component/_wikia-drawer.scss:1]: check blame

File: public/robots.txt [hunk 14]
```diff
@@ -0,0 +1,3 @@
+# http://www.robotstxt.org
```
Comment (hakubo) [inline public/robots.txt]: delete

File: public/crossdomain.xml [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
```
Comment (hakubo) [inline public/crossdomain.xml]: delete

File: app/styles/component/_wikia-drawer.scss (hunk 13 above)
Comment (hakubo) [reply app/styles/component/_wikia-drawer.scss:1]: it used to be wrapped in .side-nav
it's not anymore - so this is good to go

**Pull request merged**